File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        int result = 0;

        var set = new HashSet<int>(nums);

        foreach (var num in nums){
            if(!set.Contains(num-1)){
                int len = 0;
                while(set.Contains(num+len)){
                   len++; 
                }
                result = Math.Max(result, len);
            }     
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        result = 0
        n=len(nums)
        memo = [-1]*n

        def dfs(i):
            if(memo[i] !=-1):
                return memo[i]
            
            result = 1
            for j in range(i+1, n):
                if nums[i] < nums[j]:
                    result = max(result, 1+ dfs(j))
            memo[i] = result
            return result

        return max(dfs(i) for i in range(n))
        